pkg/requester/publicapi/endpoints_submit.go

```diff
@@ -46,12 +49,58 @@ func (s *RequesterAPIServer) submit(res http.ResponseWriter, req *http.Request)
 		return
 	}
 
-	if err := job.VerifyJobCreatePayload(ctx, &jobCreatePayload); err != nil {
+	if jobCreatePayload.ClientID == "" {
+		err := fmt.Errorf("ClientID is empty")
+		publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
+		return
+	}
+
+	if jobCreatePayload.APIVersion == "" {
+		err := fmt.Errorf("APIVersion is empty")
+		publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
+		return
+	}
+
+	apiVersion, err := model.ParseAPIVersion(jobCreatePayload.APIVersion)
+	if err != nil {
+		publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
+		return
+	}
+
+	var spec model.Spec
+	switch apiVersion {
+	case model.V1beta2:
+		var oldSpec v1beta2.Spec
+		if err := json.Unmarshal(jobCreatePayload.Spec, &oldSpec); err != nil {
+			publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
+			return
+		}
+		spec = model.ConvertV1beta2Spec(oldSpec)
+	case model.V1beta3:
+		if err := json.Unmarshal(jobCreatePayload.Spec, &spec); err != nil {
+			publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
+			return
+		}
+	default:
+		publicapi.HTTPError(ctx, res, fmt.Errorf("unhandled api version: %s", apiVersion), http.StatusInternalServerError)
+		return
+	}
+
+	// the job has been migrated to the latest version
+	submittedJob := model.Job{
+		APIVersion: model.V1beta3.String(),
+		Spec:       spec,
+	}
```

Comment:
decode the spec based on API version and convert it to the latest type. Allowing client to submit either the "current" (`v1beta2`) version of the spec, or a previous (`v1beta2`)